Demo/src/main/java/com/paypal/android/di/NetworkModule.kt

```diff
@@ -1,33 +1,91 @@
 private const val WRITE_TIMEOUT_IN_SEC = 30L
 
-val networkModule = module {
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
 
-    single {
-        initHttpClientAndInterceptor()
+    @Provides
+    fun provideRetrofitService(): Retrofit {
+        return Retrofit.Builder()
+            .baseUrl("https://ppcp-sample-merchant-sand.herokuapp.com")
+            .client(initHttpClientAndInterceptor())
+            .client(getUnsafeOkHttpClient().build())
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
     }
 
-    single {
-        GsonBuilder()
-            // Additional params
-            .create()
+    @Provides
+    fun providePaypalApi(): PayPalDemoApi {
+        return provideApi<PayPalDemoApi>(provideRetrofitService())
+    }
+
+    private fun initHttpClientAndInterceptor(): OkHttpClient {
+        val okHttpBuilder = OkHttpClient.Builder()
+        // Timeouts
+        okHttpBuilder.connectTimeout(CONNECT_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
+            .readTimeout(READ_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
+            .writeTimeout(WRITE_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
+
+        return okHttpBuilder.build()
     }
-}
 
-private fun initHttpClientAndInterceptor(): OkHttpClient {
-    val okHttpBuilder = OkHttpClient.Builder()
-    // Timeouts
-    okHttpBuilder.connectTimeout(CONNECT_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
-        .readTimeout(READ_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
-        .writeTimeout(WRITE_TIMEOUT_IN_SEC, TimeUnit.SECONDS)
+    private fun getUnsafeOkHttpClient(): OkHttpClient.Builder {
+        // Create a trust manager that does not validate certificate chains
+        val trustAllCerts: Array<TrustManager> = arrayOf(
+            @SuppressLint("CustomX509TrustManager")
+            object : X509TrustManager {
+                @Throws(CertificateException::class)
+                override fun checkClientTrusted(
+                    chain: Array<X509Certificate?>?,
+                    authType: String?
+                ) = Unit
+
+                @Throws(CertificateException::class)
+                override fun checkServerTrusted(
+                    chain: Array<X509Certificate?>?,
+                    authType: String?
+                ) = Unit
+
+                override fun getAcceptedIssuers(): Array<X509Certificate> = arrayOf()
+            }
+        )
+        // Install the all-trusting trust manager
+        val sslContext: SSLContext = SSLContext.getInstance("SSL")
+        sslContext.init(null, trustAllCerts, SecureRandom())
+        // Create an ssl socket factory with our all-trusting manager
+        val sslSocketFactory: SSLSocketFactory = sslContext.socketFactory
+        val builder = OkHttpClient.Builder()
+        builder.sslSocketFactory(
+            sslSocketFactory,
+            trustAllCerts[0] as X509TrustManager
+        )
+        builder.hostnameVerifier { _, _ -> true }
+        return builder
+    }
+}
```

Comment:
Just a small clarification about this method: since our backend repo creates a SSL server with Heroku, seems like it is not trusted or well configured, so this would be like a workaround to make it work